griptape/tools/google_docs/tool.py

```diff
@@ -0,0 +1,302 @@
+from __future__ import annotations
+from typing import TYPE_CHECKING
+import logging
+from attr import field, define
+from schema import Schema, Optional, Literal
+from griptape.artifacts import ErrorArtifact, InfoArtifact, ListArtifact, TextArtifact
+from griptape.utils.decorators import activity
+from griptape.tools import BaseGoogleClient
+
+if TYPE_CHECKING:
+    from googleapiclient.discovery import Resource
+
+@define
+class GoogleDocsClient(BaseGoogleClient):
+    DEFAULT_FOLDER_PATH = "root"
+
+    DOCS_SCOPES = ["https://www.googleapis.com/auth/documents"]
+
+    DRIVE_SCOPES = ["https://www.googleapis.com/auth/drive.file"]
+
+    owner_email: str = field(kw_only=True)
+
+    @activity(
+        config={
+            "description": "Can be used to append text to a Google Doc.",
+            "schema": Schema(
+                {
+                    Literal(
+                        "file_path",
+                        description="Destination file path in the POSIX format. For example, 'foo/bar/baz.txt'"
+                    ): str,
+                    Literal(
+                        "text",
+                        description="Text to be appended to the Google Doc."
+                    ): str
+                }
+            ),
+        }
+    )
+    def append_text_to_google_doc(self, params: dict) -> InfoArtifact | ErrorArtifact:
+        try:
+            file_path = params["file_path"]
+            text = params["text"]
+            service = self._build_client(self.DOCS_SCOPES, "docs", "v1")
+            drive_service = self._build_client(self.DRIVE_SCOPES, "drive", "v3")
+            document_id = self._path_to_file_id(drive_service, file_path)
+            if document_id:
+                doc = service.documents().get(documentId=document_id).execute()
+                content = doc["body"]["content"]
+                last_text = content[-1]["paragraph"]["elements"][-1]["textRun"]["content"]
```

Comment:
Is there always guaranteed to be a value in `doc["body"]["content"]`? Even in an empty doc?

Reply:
Yes!